File: 1845-Seat Reservation Manager.py
from heapq import *
class SeatManager:

    def __init__(self, n: int):
        self.pq = []
        for i in range(1, n+1):
            heappush(self.pq, i)

    def reserve(self) -> int:
        return heappop(self.pq)

    def unreserve(self, seatNumber: int) -> None:
        heappush(self.pq, seatNumber)

# Your SeatManager object will be instantiated and called as such:
# obj = SeatManager(n)
# param_1 = obj.reserve()
# obj.unreserve(seatNumber)

File: C++/1845-Seat Reservation Manager.cpp
class SeatManager {
    priority_queue<int, vector<int>, greater<int>> pq;  // the way to declare a min-heap
public:
    SeatManager(int n) {
        for (int i=1; i<=n; i++){
            pq.push(i);
        }
    }
    
    int reserve() {
        int res = pq.top();
        pq.pop();
        return res;
    }
    
    void unreserve(int seatNumber) {
        pq.push(seatNumber);
    }
};

/**
 * Your SeatManager object will be instantiated and called as such:
 * SeatManager* obj = new SeatManager(n);
 * int param_1 = obj->reserve();
 * obj->unreserve(seatNumber);
 */